lib/ua.rb

```diff
@@ -199,18 +199,16 @@ def ua_summary(surfaces, date = Time.now, version = "",
   b2[:ref] = blc.clone                                        #        reference
 
   # Loop through surfaces, subsurfaces and edges and populate bloc1 & bloc2.
-  surfaces.values.each do |surface|
+  surfaces.each do |id, surface|
     next unless surface.has_key?(:deratable)
     next unless surface[:deratable]
     next unless surface.has_key?(:type)
     type = surface[:type]
     next unless type == :wall || type == :ceiling || type == :floor
     next unless surface.has_key?(:net)
     next unless surface[:net] > TOL
-    next unless surface.has_key?(:u) || surface.has_key?(:r)
-    next unless surface[:u] > TOL if surface.has_key?(:u)
-    next unless surface[:r] > TOL if surface.has_key?(:r)
-    surface[:u] = 1.0 / surface[:r] unless surface.has_key?(:u)
```

Comment:
Another oversight. The `:r` variable strictly refers to the targeted insulation material (to derate), not the assembly as a whole.